pkg/controller/installation/core_controller.go

```diff
@@ -906,10 +906,17 @@ func (r *ReconcileInstallation) Reconcile(ctx context.Context, request reconcile
 	}
 
 	var esAdminSecret *v1.Secret
+	var kubeControllersUserSecret *v1.Secret
 	if instance.Spec.Variant == operator.TigeraSecureEnterprise {
-		// Kube controllers needs the admin secret copied to it's namespace as it has administrative tasks to run on
-		// Elasticsearch.
-		esAdminSecret, err = utils.GetSecret(ctx, r.client, render.ElasticsearchAdminUserSecret, rmeta.OperatorNamespace())
+		// Kube controllers no longer uses the admin secret to run it's administrative tasks on Elasticsearch.
+		// Instead a username/password combo is randomly generated and stored in tigera-ee-kube-controllers-elasticsearch-access
+		// and passed to the ES Gateway deployment to be used to detect requests from kube-controllers and swap in admin
+		// level Elasticsearch credentials.
+		kubeControllersUserSecret, err = utils.GetSecret(ctx, r.client, render.ElasticsearchKubeControllersUserSecret, common.CalicoNamespace)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+		esAdminSecret, err = utils.GetSecret(ctx, r.client, render.ElasticsearchAdminUserSecret, render.ElasticsearchNamespace)
```

Comment:
Reading the comment on kubeControllersUserSecret, do we still need this?

Reply:
Yes we do, I forgot to update the comment since we wen't back and forth on this a few times. Initially we planned to eliminate the admin user secret for kube controllers entirely and have a special case in the gateway that would swap in the admin credentials that the gateway already has access to on all kube-controllers Elasticsearch requests. But later we decided to eliminate the special case and have the same dummy credential -> real credential swap for kube-controllers as we do for other components. So we still need to create an admin user secret for kube-controllers, the only difference being that we create it in the tigera-elasticsearch namespace now instead of calico-system. 

I will update the comment to reflect this.

Reply:
Actually I will remove this comment and write up something similar in kube-controllers.go

Reply:
we are only giving kube controllers the dummy set, the admin set is being dropped into the tigera-elasticsearch namespace and only accessed by es-gateway during the swapping.

Reply:
The kube controllers controller should not be responsible for copying to the elasticsearch namespace, that should be handled by the logstorage controller.